.changelog/24974.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+state store: fix for setting correct job status in various scenarios
```

Comment:
We should probably try to summarize what those scenarios are as best we can here.